.doc_gen/readmes/README.md

```diff
@@ -0,0 +1,103 @@
+
+```
+python -m venv venv
+venv/Scripts/activate
+python -m pip install -r requirements.txt
+```
+
+Depending on how you have Python installed and on your operating system,
+the commands may vary slightly, for example on Windows use `py` in place of
+`python` and uses backslashes in the `venv` path.
+
+## Generate a README
+
+WRITEME creates content primarily from metadata you have already
+authored for the SOS project. After you have authored metadata and snippet tags
+for your examples, run the following command in the root folder of the repo:
+
+```
+python .doc_gen/readmes/writeme.py <language> <version> <service>
+```
+
+WRITME reads metadata and config data and generates a README in the service
+folder for the specified language, version, and service.
+
+For example, to generate an S3 README for Python:
+
+```
+python .doc_gen/readmes/writeme.py Python 3 s3
+```
+
+This creates a README.md file in the `python/example_code/s3` folder.
+
+### Parameters
+
+* `language` must match a top-level language in sdks.yaml.
+* `version` must be defined for the language in sdks.yaml.
+* `service` must match a top-level service in services.yaml.
+* `--readme` (optional) defines an alternate for the output README (default is README.md).
+* `--svc_folder` (optional) overrides the output folder for the README.
+
+You can get inline usage info by using the `-h` flag:
+
+```
+python .doc_gen/readmes/writeme.py -h
+``` 
+
+### Configuration
+
+Additional configuration is kept in `config.py`.
+
+* `entities` is a dictionary of entities that are not otherwise defined in
+services.yaml.
+* `language` is a dictionary of language and version for each SDK. Fields are:
+    * `base_folder` the root folder for the SDK version.
+    * `service_folder` a Jinja template of the service folder for the SDK version.
+    This may not work in all cases. If not, use the `--svc_folder` override.
+    * `sdk_api_ref` a Jinja template of the SDK API Reference topic for the SDK version.
+    This is used to create the link to the reference page in the Additional Resources
+    section, such as to the Boto3 S3 reference page for Python.
+    
+### Custom content
+
+Custom content can be per-SDK or per-README.
+
+#### SDK custom content
+
+Currently, you can define per-SDK content in `includes/prerequisites.jinja2` and
+`includes/run_instructions.jinja2`. In each of these files, you can find (or add) `if`
+blocks that include custom content for specific SDK versions.
+
+#### README custom content
+
+Each README can have custom content in specific places. The first time you
+generate a README, it contains empty blocks designated by special comments, such as
+
+```
+<!--custom.prerequisites.start-->
+<!--custom.prerequisites.end-->
+```
+
+Any content you add within these comments is preserved in subsequent generations
+of the README. Do not change the names of these comments or remove them! Just leave them
+empty if you don't have need for custom content.
```

Comment:
Update to final two sentences: 
Do not change the names of these comments or remove them. Keep them
empty if you don't need custom content.

Reply:
fixed